src/components/CartPage/CartPage.jsx

```diff
@@ -0,0 +1,144 @@
+import styled from 'styled-components';
+import CartHeader from 'components/CartPage/CartHeader';
+import CartProduct from 'components/CartPage/CartProduct';
+import Order from 'components/CartPage/Order';
+import CheckBox from 'components/CartPage/CheckBox';
+import { DivideUnderLine, DefaultButton } from 'components/shared/styles';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { deleteCart } from 'store/carts/action';
+import { ERROR_MESSAGE, NOTICE } from 'constants';
+
+function CartPage() {
+  const { carts } = useSelector((state) => state.carts);
+
+  const dispatch = useDispatch();
+
+  const [checkedList, setCheckedList] = useState(carts);
+
+  const updateCheckedList = (product, isChecked) => {
+    if (isChecked) {
+      setCheckedList(checkedList.filter((item) => item.id !== product.id));
+      return;
+    }
+    const newList = [...checkedList];
+
+    newList.push(product);
+    setCheckedList(newList);
```

Comment:
무작정 짧게 코딩하는 걸 선호하진 않지만, 아래와 같이 하는 게 더 직관적인 것 같은데 어떻게 생각하시나요?!

```suggestion
    setCheckedList([...checkedList, product])
```

Reply:
제가 작성한 코드를 보니 데이터 불변성에 대한 확신이 없는 상태에서 코드를 작성한 것 같은 느낌이 드네요... 당연히 spread 연산을 통해서 한 줄에 끝낼 수 있는 코드였던 것 같습니다. 월터님이 제안해주신 코드로 수정하는 것이 훨씬 좋은 것 같습니다 ㅎㅎㅎ

Reply:
이 부분은 CartPage에서 관리하던 checkedList state를 제거하고 store에 있는 carts 에 checked 키값을 추가해서 store에서 관리하는 방식으로 수정하는 과정에서 삭제됐습니다.

[해당커밋](https://github.com/woowacourse/react-shopping-cart/pull/142/commits/8c6bf2f0d35245dd4b145682bb0cc963f2136a0a)